src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -1,19 +1,45 @@
 (ns puppetlabs.puppetdb.http.command
   (:require [puppetlabs.puppetdb.command.constants :refer [command-names]]
             [puppetlabs.trapperkeeper.core :refer [defservice]]
+            [clojure.string :as str]
             [clojure.tools.logging :as log]
             [puppetlabs.puppetdb.command :as command]
             [puppetlabs.puppetdb.http :as http]
+            [puppetlabs.puppetdb.cheshire :as json]
             [net.cgrand.moustache :as moustache]
             [puppetlabs.puppetdb.middleware :as mid]
             [compojure.core :as compojure]))
 
+(def min-supported-commands
+  {"replace catalog" 6
+   "replace facts" 4
+   "store report" 5
+   "deactivate node" 3})
```

Comment:
Is there anyway to collect the map of supported versions from the `defmulti` @mullr @senior? I've never seen anything like that but seems like you should be able to check everything a defmethod is valid for.